### ISSUE TITLE
days_until_first_cert_expiring does not follow documentation.

### ISSUE THREAD

**Issue by ashimrana** [envoyproxy/envoy#5442]
days_until_first_cert_expiring does not follow documentation.

As per the documentation days_until_first_cert_expiring supposed to report the number of days until the next certificate being managed will expire.

I set up a server validation_context for trusted_ca and listed a couple of server certificates with their whole chain up there. It seems that days_until_first_cert_expiring does not reorder as per certificate validity days and just reports the validity of the first certificate present in the the ca file.

Also could not understand the next part in the "next certificate being managed will expire" statement. It seems that even though the first certificate present has expired, envoy proxy does not report the expiry of the other certificates.

/certs API also does not list all certificates present in trusted_ca file.

However there does not seems to be any issues with validation functionality as such.

**Comment by ramaraochavali** [comment]
/certs endpoint displays only the first certificate. There is an enhancement https://github.com/envoyproxy/envoy/issues/4661 to display all.

**Comment by ashimrana** [comment]
I observed that if I create a trusted_ca file with the self signed certificates of all the hosts then except the first host all others requests fails with SSL error: 268435581:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED. 

Not sure how to make envoy trust all of the certificates. If I use the same CA file for curl, requests to all of the hosts succeed. 

Note that for testing purpose I have used same ip but different ports to create a cluster.
